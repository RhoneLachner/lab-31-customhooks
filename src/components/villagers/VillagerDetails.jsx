import React from 'react';
import PropTypes from 'prop-types';
import styles from '../containers/containers.css';

const Villager = ({ name, image, quote, skill, birthday  }) => (
  <figure className={styles.detailItem}>
    <img src={image} alt={name}></img>
    <figcaption>
      <p>Name: {name}</p>
      <p>Quote: {quote}</p>
      <p>Skill: {skill}</p>
      <p>Birthday: {birthday}</p>
    </figcaption>

  </figure>
);

Villager.propTypes = {
  name: PropTypes.string.isRequired,
  image: PropTypes.string.isRequired,
  quote: PropTypes.string.isRequired,
  skill: PropTypes.string.isRequired,
  birthday: PropTypes.string.isRequired
};

export default Villager;
