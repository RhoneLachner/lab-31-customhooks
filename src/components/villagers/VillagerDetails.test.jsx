import React from 'react';
import { render, cleanup } from '@testing-library/react';
import VillagerDetails from './VillagerDetails';

describe('VillagerDetails component', () => {
  afterEach(() => cleanup());
  it('renders VillagerDetails', () => {
    const { asFragment } = render(<VillagerDetails />);
    expect(asFragment()).toMatchSnapshot();
  });
});
