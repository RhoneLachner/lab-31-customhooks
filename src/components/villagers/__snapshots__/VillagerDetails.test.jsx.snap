// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`VillagerDetails component renders VillagerDetails 1`] = `
<DocumentFragment>
  <figure
    class="detailItem"
  >
    <img />
    <figcaption>
      <p>
        Name: 
      </p>
      <p>
        Quote: 
      </p>
      <p>
        Skill: 
      </p>
      <p>
        Birthday: 
      </p>
    </figcaption>
  </figure>
</DocumentFragment>
`;
